domain-services/fiscal-event-post-processor/src/main/java/org/egov/consumer/FiscalEventDereferenceConsumer.java

```diff
@@ -0,0 +1,45 @@
+package org.egov.consumer;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.egov.config.FiscalEventPostProcessorConfig;
+import org.egov.producer.Producer;
+import org.egov.service.FiscalEventUnbundleService;
+import org.egov.web.models.FiscalEventDeReferenced;
+import org.egov.web.models.FiscalEventLineItemUnbundled;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+
+@Slf4j
+@Component
+public class FiscalEventDereferenceConsumer {
+
+    @Autowired
+    private FiscalEventUnbundleService unbundleService;
+
+    @Autowired
+    private FiscalEventPostProcessorConfig processorConfig;
+
+    @Autowired
+    private Producer producer;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @KafkaListener(topics = {"${fiscal.event.kafka.dereferenced.topic}"})
+    public void listen(final HashMap<String, Object> record, @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        try {
+            FiscalEventDeReferenced fiscalEventDeReferenced = mapper.convertValue(record, FiscalEventDeReferenced.class);
+            List<FiscalEventLineItemUnbundled> fiscalEventLineItemUnbundledList  = unbundleService.unbundle(fiscalEventDeReferenced);
+            producer.push(processorConfig.getFiscalEventDruidTopic(), fiscalEventLineItemUnbundledList);
```

Comment:
There will also be another producer.push to the flattened topic. 
Both producer.push will come after we have made the flatten function call.

Reply:
Yes we know..once we'll do flatten. will do the changes.